src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -142,15 +142,15 @@ public async Task<(Error error, GitHubUser user)> GetByLogin(string login)
             return (null, TryGetByLogin(login));
         }
 
-        public async Task<(Error, GitHubUser)> GetByCommit(string authorEmail, string repoOwner, string repoName, string commitSha)
+        public async Task<(Error, GitHubUser)> GetByEmailOrCommit(string authorEmail, string repoOwner, string repoName, string commitSha)
         {
             if (string.IsNullOrEmpty(authorEmail))
                 return default;
 
             Telemetry.TrackCacheTotalCount(TelemetryName.GitHubUserCache);
             var user = TryGetByEmail(authorEmail);
-            if (user != null)
-                return (null, user.IsValid() ? user : null);
+            if (user != null || string.IsNullOrEmpty(repoOwner) || string.IsNullOrEmpty(repoName))
+                return (null, (user?.IsValid() ?? false) ? user : null);
```

Comment:
consider adding a new method `GetByEmail` that only calls `TryGetByEmail` and leave `GetByCommit` intact.

Reply:
this may change the first condition trigger, need confirmed with @superyyrrzz.